presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -95,11 +110,34 @@
             .build(CacheLoader.from(key ->
                     internalCompileHashStrategy(key.getTypes(), key.getOutputChannels(), key.getJoinChannels(), key.getSortChannel())));
 
+    private final LoadingCache<Type, ScalarFunctionImplementation> distinctFromOperatorCache = CacheBuilder.newBuilder()
```

Comment:
Why do we introduce this cache? 

cc @haozhun

Reply:
It is more expensive to get from FunctionRegistry, I can remove this if it's an unnecessary optimization.

Reply:
@elonazoulay : If we want to cache results to `resolveFunction`, we should cache it in `FunctionRegistry` directly :). https://github.com/prestodb/presto/pull/10381